tools/run_tests/python_utils/upload_test_results.py

```diff
@@ -114,3 +127,41 @@ def upload_results_to_bq(resultset, bq_table, args, platform):
           else:
             print('Error uploading result to bigquery, all attempts failed.')
             sys.exit(1)
+
+
+def upload_interop_results_to_bq(resultset, bq_table, args):
+  """Upload interop test results to a BQ table.
+
+  Args:
+      resultset: dictionary generated by jobset.run
+      bq_table: string name of table to create/upload results to in BQ
+      args: args in run_interop_tests.py, generated by argparse
+  """
+  bq = big_query_utils.create_big_query()
+  big_query_utils.create_partitioned_table(bq, _PROJECT_ID, _DATASET_ID, bq_table, _INTEROP_RESULTS_SCHEMA, _DESCRIPTION,
+                                           partition_type=_PARTITION_TYPE, expiration_ms= _EXPIRATION_MS)
+
+  for shortname, results in six.iteritems(resultset):
+    for result in results:
+      test_results = {}
+      _get_build_metadata(test_results)
+      test_results['elapsed_time'] = '%.2f' % result.elapsed_time
+      test_results['result'] = result.state
+      test_results['test_name'] = shortname
+      test_results['suite'] = shortname.split(':')[0]
+      test_results['client'] = shortname.split(':')[1]
+      test_results['server'] = shortname.split(':')[2]
+      test_results['test_case'] = shortname.split(':')[3]
+      test_results['timestamp'] = time.strftime('%Y-%m-%d %H:%M:%S')
+      row = big_query_utils.make_row(str(uuid.uuid4()), test_results)
+      # TODO(jtattermusch): rows are inserted one by one, very inefficient
+      max_retries = 3
```

Comment:
perhaps some refatoring would be useful to avoid copy-pasting the upload-with-retries code. Ideally we would address this together with the "TODO(jtattermusch): rows are inserted one by one, very inefficient", so fine to leave the code as is in the PR and look into refactoring later.